sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -224,9 +224,10 @@ namespace Azure { namespace Core { namespace Http {
     Azure::Core::IO::BodyStream* GetBodyStream() { return this->m_bodyStream; }
 
     /**
-     * @brief A value indicating whether download is happening via stream.
+     * @brief A value indicating whether download will return the raw response within a memory buffer or if it will provide a
+     * body stream instead.
      */
-    bool IsDownloadViaStream() { return this->m_isDownloadViaStream; }
+    bool IsBufferedDownload() { return this->m_isBufferedDownload; }
```

Comment:
Just for ~2-3 APIs in one service, we have elevated the concept of "download" for all request objects by naming the bool like this.

I know naming is difficult, but should we consider something like the following? Basically convey the intent exactly, and not be specific to a storage operation. I don't think we should use the term `download` in the `request<->response` interaction. It's coupling concepts too tightly, especially when storage sits on top of core.

```suggestion
    bool IsResponseBuffered() { return this->m_isBufferedDownload; }
```
OR
```suggestion
    bool ShouldBufferResponse() { return this->m_isBufferedDownload; }
```
Thoughts?

Reply:
In the future, a good place to add `IsResponseBuffered` would be on the `RawResponse` itself.

So, the pair becomes `Request::ShouldBufferResponse()` and `RawRespnose::IsResponseBuffered()`.

cc @JeffreyRichter, @RickWinter